packages/transactions/src/keys.ts

```diff
@@ -111,64 +113,62 @@ export function publicKeyFromSignatureRsv(
+  return length === PRIVATE_KEY_COMPRESSED_LENGTH;
 }
 
-export function publicKeyToString(key: StacksPublicKey): string {
-  return bytesToHex(key.data);
+export function publicKeyIsCompressed(publicKey: PublicKey): boolean {
+  return !publicKeyToHex(publicKey).startsWith('04');
 }
 
 export function serializePublicKey(key: StacksPublicKey): Uint8Array {
   return key.data.slice();
 }
 
-export function pubKeyfromPrivKey(privateKey: string | Uint8Array): StacksPublicKey {
-  const privKey = createStacksPrivateKey(privateKey);
-  const publicKey = nobleGetPublicKey(privKey.data.slice(0, 32), privKey.compressed);
-  return createStacksPublicKey(bytesToHex(publicKey));
+/**
+ * Get the public key from a private key.
+ * Allows for "compressed" and "uncompressed" private keys.
+ * > Matches legacy `pubKeyfromPrivKey`, `getPublic` function behavior
+ */
+export function privateKeyToPublic(privateKey: PrivateKey): string {
+  privateKey = privateKeyToBytes(privateKey);
+  const isCompressed = privateKeyIsCompressed(privateKey);
+  return bytesToHex(nobleGetPublicKey(privateKey.slice(0, 32), isCompressed));
 }
 
-export function compressPublicKey(publicKey: string | Uint8Array): StacksPublicKey {
-  const hex = typeof publicKey === 'string' ? publicKey : bytesToHex(publicKey);
-  const compressed = Point.fromHex(hex).toHex(true);
-  return createStacksPublicKey(compressed);
+export function compressPublicKey(publicKey: PublicKey): string {
+  return Point.fromHex(publicKeyToHex(publicKey)).toHex(true);
 }
 
 export function deserializePublicKey(bytesReader: BytesReader): StacksPublicKey {
   const fieldId = bytesReader.readUInt8();
   const keyLength =
     fieldId === 4 ? UNCOMPRESSED_PUBKEY_LENGTH_BYTES : COMPRESSED_PUBKEY_LENGTH_BYTES;
-  return publicKeyFromBytes(concatArray([fieldId, bytesReader.readBytes(keyLength)]));
+  return createStacksPublicKey(concatArray([fieldId, bytesReader.readBytes(keyLength)]));
 }
 
-export interface StacksPrivateKey {
-  // "compressed" private key is a misnomer: https://web.archive.org/web/20220131144208/https://www.oreilly.com/library/view/mastering-bitcoin/9781491902639/ch04.html#comp_priv
-  // it actually means: should public keys be generated as "compressed" or "uncompressed" from this private key
-  compressed: boolean;
-  data: Uint8Array;
+// todo: double-check for deduplication, rename!
+export function makeRandomPrivKey(): string {
+  return bytesToHex(utils.randomPrivateKey());
 }
 
-export function createStacksPrivateKey(key: string | Uint8Array): StacksPrivateKey {
-  const data = privateKeyToBytes(key);
-  const compressed = data.length == PRIVATE_KEY_COMPRESSED_LENGTH;
-  return { data, compressed };
-}
-
-export function makeRandomPrivKey(): StacksPrivateKey {
-  return createStacksPrivateKey(utils.randomPrivateKey());
-}
+// todo: complete refactor
+export type PrivateKey = string | Uint8Array;
+export type PublicKey = string | Uint8Array;
```

Comment:
Do you think it's worth keeping around some backward compatibility? For example, functions that accept a key as an arg could use something like
```ts
// Allow the `StacksPrivateKey` type for backwards compat
export type PrivateKeyArg = string | Uint8Array | StacksPrivateKey;
```

Though maybe that's too much baggage to carry around